### PR TITLE
Bump error handling to `failure`

### PR DESCRIPTION
**Pull request by fominok**
Tried to do migration for `failure` crate, as proposed in #60

### CODE REVIEW

**Comment by patrickisgreene** [comment]
Thank you for this! I have been way to busy to get back to working on Xi-term 😄

On another note could you update clippy to version 0.0.196, that should fix travis, also ill leave this open for a little while to give @little-dude & @ryukinix time to look over it

**Comment by fominok** [comment]
Seems it builds now. Waiting for review, thanks!

**Review by ryukinix — APPROVED** [review]
Seems a nice improvement. `failure` crates has a better legibility than `error-chain`.

**Comment by fominok** [comment]
Hey, still needing something?

**Comment by little-dude** [comment]
Sorry. @ryukinix @bytebuddha I think we don't need the 3 of us approving for these PRs. I tend to be quite un-responsive for variable period of times and I don't want people to be frustrated waiting for me :S

File: src/main.rs [hunk 1]
```diff
@@ -99,7 +94,7 @@ fn run() -> Result<()> {
     }
 
     info!("starting the event loop");
-    let mut core = Core::new().chain_err(|| "failed to create event loop")?;
+    let mut core = Core::new().context("Failed to create event loop")?;
```
Comment (little-dude) [inline src/main.rs:97]: Oh `failure` is really nice :) This closure always felt a bit weird.

**Review by little-dude — APPROVED** [review]
Nice, thank you :)

**Comment by little-dude** [comment]
thanks @fominok and sorry for the delay!

**Pull request merged**